### PR TITLE
Changing WorkflowRun in order to run FlowExecutionListener.fireComple…

### PR DESCRIPTION
**Pull request by joseblas**
…ted just after unListener.fireCompleted and before closing the listener

Added some extra work in the test to make sure that some actions are allowed

### CODE REVIEW

**Comment by joseblas** [comment]
related to https://github.com/jenkinsci/workflow-job-plugin/pull/142/files

File: src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunWithFlowExecutionListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        public void onCompleted(@Nonnull FlowExecution execution) {
+            super.onCompleted(execution);
+            Queue.Executable executable = null;
+            try {
+                PrintStream logger = execution.getOwner().getListener().getLogger();
+                logger.println("blah blah blah");
+                executable = execution.getOwner().getExecutable();
+                if( executable instanceof WorkflowRun){
+                    WorkflowRun run = (WorkflowRun) executable;
+                    Action action = new Action() {
```
Comment (dwnusbaum) [inline src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunWithFlowExecutionListenerTest.java]: nit: It doesn't look like the test uses this action anywhere, and it doesn't seem related to when the listener fires, so I'd just delete it to keep the test simple.

File: src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunWithFlowExecutionListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                            return null;
+                        }
+                    };
+                    run.addAction(action);
+                    WorkflowJob workflowJob = run.getParent();
+                    int number = workflowJob.getLastBuild().number;
+                    logger.println("Build Number :" + number);
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (dwnusbaum) [inline src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunWithFlowExecutionListenerTest.java]: I would rethrow this as an unchecked exception instead so that the build and test would fail if there was a problem:

```suggestion
                throw new RuntimeException(e);
```
